Cleanup the generation of the nix3 manpages

regnat opened the pull request:
Use a dedicated make target for each man page rather than bundling the
generation as part of `install`.

This has two advantages:
- The generation is properly cached and we get rid of the mysterious `GEN
  install` when running `make install`
- Everything is properly parallel (not that the doc generation is the
  bottleneck in general, but it still takes a few hundred miliseconds
  that are nice to avoid for incremental builds)

edolstra commented:
Looks like I need to run `make install` three times to reach a fixed point:
```
$ touch src/nix/develop.cc 

$ make install -j16        
  CXX    src/nix/develop.o
  LD     /home/eelco/Dev/nix/outputs/out/bin/nix
  GEN    doc/manual/conf-file.json
  GEN    doc/manual/builtins.json
  GEN    doc/manual/nix.json
  GEN    doc/manual/src/command-ref/conf-file.md
  GEN    doc/manual/src/expressions/builtins.md
  GEN    doc/manual/src/command-ref/new-cli
  GEN    doc/manual/nix.conf.5
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man5/nix.conf.5
  GEN    doc/manual/src/SUMMARY.md
  GEN    /home/eelco/Dev/nix/outputs/doc/share/doc/nix/manual/index.html

$ make install -j16 
  GEN    doc/manual/new-cli/nix3-daemon.1
  GEN    doc/manual/new-cli/nix3-build.1
  GEN    doc/manual/new-cli/nix3-bundle.1
  GEN    doc/manual/new-cli/nix3-copy.1
  GEN    doc/manual/new-cli/nix3-edit.1
  GEN    doc/manual/new-cli/nix3-doctor.1
  GEN    doc/manual/new-cli/nix3-describe-stores.1
  GEN    doc/manual/new-cli/nix3-develop.1
  GEN    doc/manual/new-cli/nix3-flake-check.1
  GEN    doc/manual/new-cli/nix3-eval.1
  GEN    doc/manual/new-cli/nix3-flake-clone.1
  GEN    doc/manual/new-cli/nix3-flake-archive.1
  GEN    doc/manual/new-cli/nix3-flake-init.1
  GEN    doc/manual/new-cli/nix3-flake-lock.1
  GEN    doc/manual/new-cli/nix3-flake-info.1
  GEN    doc/manual/new-cli/nix3-flake.1
  GEN    doc/manual/new-cli/nix3-flake-metadata.1
  GEN    doc/manual/new-cli/nix3-flake-prefetch.1
  GEN    doc/manual/new-cli/nix3-flake-show.1
  GEN    doc/manual/new-cli/nix3-flake-new.1
  GEN    doc/manual/new-cli/nix3-flake-update.1
  GEN    doc/manual/new-cli/nix3-hash-file.1
  GEN    doc/manual/new-cli/nix3-hash.1
  GEN    doc/manual/new-cli/nix3-hash-path.1
  GEN    doc/manual/new-cli/nix3-hash-to-base16.1
  GEN    doc/manual/new-cli/nix3-hash-to-base32.1
  GEN    doc/manual/new-cli/nix3-hash-to-base64.1
  GEN    doc/manual/new-cli/nix3-hash-to-sri.1
  GEN    doc/manual/new-cli/nix3-help.1
  GEN    doc/manual/new-cli/nix3-key-convert-secret-to-public.1
  GEN    doc/manual/new-cli/nix3-key-generate-secret.1
  GEN    doc/manual/new-cli/nix3-key.1
  GEN    doc/manual/new-cli/nix3-log.1
  GEN    doc/manual/new-cli/nix3-nar-dump-path.1
  GEN    doc/manual/new-cli/nix3-nar-cat.1
  GEN    doc/manual/new-cli/nix3-nar.1
  GEN    doc/manual/new-cli/nix3-nar-ls.1
  GEN    doc/manual/new-cli/nix3-print-dev-env.1
  GEN    doc/manual/new-cli/nix3-path-info.1
  GEN    doc/manual/new-cli/nix3-profile-history.1
  GEN    doc/manual/new-cli/nix3-profile-diff-closures.1
  GEN    doc/manual/new-cli/nix3-profile-list.1
  GEN    doc/manual/new-cli/nix3-profile-install.1
  GEN    doc/manual/new-cli/nix3-profile.1
  GEN    doc/manual/new-cli/nix3-profile-remove.1
  GEN    doc/manual/new-cli/nix3-realisation-info.1
  GEN    doc/manual/new-cli/nix3-profile-upgrade.1
  GEN    doc/manual/new-cli/nix3-realisation.1
  GEN    doc/manual/new-cli/nix3-registry-add.1
  GEN    doc/manual/new-cli/nix3-registry-list.1
  GEN    doc/manual/new-cli/nix3-registry.1
  GEN    doc/manual/new-cli/nix3-registry-pin.1
  GEN    doc/manual/new-cli/nix3-registry-remove.1
  GEN    doc/manual/new-cli/nix3-run.1
  GEN    doc/manual/new-cli/nix3-repl.1
  GEN    doc/manual/new-cli/nix3-show-config.1
  GEN    doc/manual/new-cli/nix3-store-add-file.1
  GEN    doc/manual/new-cli/nix3-shell.1
  GEN    doc/manual/new-cli/nix3-store-add-path.1
  GEN    doc/manual/new-cli/nix3-store-delete.1
  GEN    doc/manual/new-cli/nix3-store-copy-sigs.1
  GEN    doc/manual/new-cli/nix3-show-derivation.1
  GEN    doc/manual/new-cli/nix3-store-cat.1
  GEN    doc/manual/new-cli/nix3-search.1
  GEN    doc/manual/new-cli/nix3-store-diff-closures.1
  GEN    doc/manual/new-cli/nix3-store-dump-path.1
  GEN    doc/manual/new-cli/nix3-store-gc.1
  GEN    doc/manual/new-cli/nix3-store-make-content-addressable.1
  GEN    doc/manual/new-cli/nix3-store-ls.1
  GEN    doc/manual/new-cli/nix3-store.1
  GEN    doc/manual/new-cli/nix3-store-ping.1
  GEN    doc/manual/new-cli/nix3-store-optimise.1
  GEN    doc/manual/new-cli/nix3-store-repair.1
  GEN    doc/manual/new-cli/nix3-store-prefetch-file.1
  GEN    doc/manual/new-cli/nix3-store-sign.1
  GEN    doc/manual/new-cli/nix3-store-verify.1
  GEN    doc/manual/new-cli/nix3-upgrade-nix.1
  GEN    doc/manual/new-cli/nix3-why-depends.1
  GEN    doc/manual/new-cli/nix.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-build.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-bundle.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-copy.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-daemon.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-describe-stores.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-develop.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-doctor.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-edit.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-eval.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-archive.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-check.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-info.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-init.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-clone.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-lock.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-metadata.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-new.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-prefetch.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-show.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-flake-update.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-hash-file.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-hash.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-hash-path.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-hash-to-base16.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-hash-to-base32.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-hash-to-base64.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-hash-to-sri.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-help.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-key-convert-secret-to-public.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-key-generate-secret.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-key.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-log.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-nar-cat.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-nar-dump-path.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-nar-ls.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-nar.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-path-info.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-profile-history.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-print-dev-env.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-profile-diff-closures.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-profile-install.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-profile-list.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-profile.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-profile-remove.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-profile-upgrade.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-realisation-info.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-realisation.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-registry.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-registry-add.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-registry-list.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-registry-remove.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-registry-pin.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-repl.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-run.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-search.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-shell.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-show-config.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-show-derivation.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-add-file.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-add-path.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-cat.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-delete.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-copy-sigs.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-diff-closures.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-dump-path.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-gc.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-ls.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-make-content-addressable.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-optimise.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-ping.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-repair.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-prefetch-file.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-sign.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-store-verify.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-upgrade-nix.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix3-why-depends.1
  INST   /home/eelco/Dev/nix/outputs/out/share/man/man1/nix.1

$ make install -j16 
  GEN    doc/manual/src/SUMMARY.md
  GEN    /home/eelco/Dev/nix/outputs/doc/share/doc/nix/manual/index.html

$ make install -j16 
make: Nothing to be done for 'install'.
```

regnat commented:
> Looks like I need to run make install three times to reach a fixed point:

Duh. OK, let me have a look

regnat commented:
Oh I see, the wildcard to list all the nix3 manpage srcs was wrong as they are generated files (so they aren’t seen on the first time). I’ve reverted that part to keep the generation of all the manpages as one make target (looping over all the md files in the directory).

I’ve also fixed a couple of things that caused spurious rebuilds

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/tweag-nix-dev-update-16/14379/1
